src/cluster_legacy.c

```diff
@@ -2042,6 +2042,41 @@ static void getClientPortFromGossip(clusterMsgDataGossip *g, int *tls_port, int
     }
 }
 
+/* Returns a string with the byte representation of the node ID (i.e. nodename)
+ * along with 8 trailing bytes for debugging purposes. */
+char *getCorruptedNodeIdByteString(clusterMsgDataGossip *gossip_msg) {
+    const int num_bytes = CLUSTER_NAMELEN + 8;
+    /* Allocate enough room for 4 chars per byte + null terminator */
+    char *byte_string = (char*) zmalloc((num_bytes*4) + 1); 
+    const char *name_ptr = gossip_msg->nodename;
+
+    /* Ensure we won't print beyond the bounds of the message */
+    serverAssert(name_ptr + num_bytes <= (char*)gossip_msg + sizeof(clusterMsgDataGossip));
+
+    for (int i = 0; i < num_bytes; i++) {
+        snprintf(byte_string + 4*i, 5, "\\x%02hhX", name_ptr[i]);
+    }
```

Comment:
Our static analyzer is complaining that this a memory out of bounds (checking 47 bytes of 40). Although we are properly checking this, that means other static analyzer tools will probably complain. Maybe we explicitly print out the two fields after nodename.

Reply:
I liked that this code would work regardless of the `clusterMsgDataGossip` internal structure, but assuming it's unlikely to change, explicitly printing out the two fields after is fine

Reply:
I think I'll just mark this as a false positive for now.